GCXMulticastDNSKit/Discovery.swift

```diff
@@ -90,7 +90,7 @@ public enum DiscoveryError: Int, Error {
 
 
 /// the protocol used for the delegate
-public protocol DiscoveryDelegate {
+public protocol DiscoveryDelegate: class {
```

Comment:
Where ist the benefit restricting protocol adoption to class types only? We would exclude structs from usage for example.

Reply:
Otherwise you can't declare the `delegate` property as weak in the `Discovery` class:
_'weak' may only be applied to class and class-bound protocol types_

Reply:
Yeah you are right.